src/woo_search/search_index/api/viewsets/search.py

```diff
@@ -0,0 +1,112 @@
+from django.utils.translation import gettext_lazy as _
+
+from drf_spectacular.utils import extend_schema
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from woo_search.search_index.api.serializers import (
+    SearchResponseSerializer,
+    SearchSerializer,
+)
+
+from ...client import get_client
+from ...constants import SortChoices
+from ...typing import SearchType
+
+
+@extend_schema(
+    tags=["index"],
+    summary=_("Search"),
+    description=_("Search the publication and document records."),
+    responses={
+        200: SearchResponseSerializer(many=True),
+    },
+)
+class SearchView(APIView):
+    serializer_class = SearchSerializer
+    response_serializer_class = SearchResponseSerializer
+
+    def _query_multi_match(self, query: str):
+        return {
+            "multi_match": {
+                "fields": [
+                    "officiele_titel^2",
+                    "verkorte_titel^1.5",
+                    "omschrijving",
+                    "identifier^3",
+                ],
+                "fuzziness": 1,
+                "query": query,
+            }
+        }
+
+    def _build_must(self, params: SearchType):
+        must = []
+
+        if query := params.get("query"):
+            must.append(self._query_multi_match(query))
+
+        return must
+
+    def base_search_query(self, params: SearchType):
+        return {
+            "index": params.get("result_type"),
+            "size": params.get("page_size"),
+            "from": params.get("page"),
+            "track_total_hits": True,
+        }
+
+    def search_query(self, params: SearchType):
+        search = self.base_search_query(params)
+
+        if sort := params.get("sort"):
+            sort_order = [
+                {"_score": {"order": "desc"}},
+                {"laatst_gewijzigd_datum": {"order": "desc"}},
+            ]
+
+            match sort:
+                case SortChoices.chronological:
+                    search["sort"] = sort_order[::-1]
+                case SortChoices.relevance:
+                    search["sort"] = sort_order
```

Comment:
I dislike this `.reverse` or `[::-1]` syntax, it makes me having to think about what's happening instead of just being able to read what the result is. I'd much prefer just separate assignments.

If we _were_ to use elasticsearch-dsl, this would instead become something like:

```py
s = Search()
match sort:
    case SortChoices.relevance:
        s = s.sort("-_score", "-laatst_gewijzigd_datum")
    case SortChoices.chronological:
        s = s.sort("-laatst_gewijzigd_datum", "-_score")
```
and I quite like the readability of that